lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -835,14 +851,26 @@ def network_tokenization?(payment_method)
 
       def subsequent_nt_apple_pay_auth(source, options)
         return unless options[:stored_credential] || options[:stored_credential_overrides]
-        return unless @@payment_solution[source]
+        return unless @@wallet_payment_solution[source]
 
         options.dig(:stored_credential_overrides, :subsequent_auth) || options.dig(:stored_credential, :initiator) == 'merchant'
       end
 
       def add_auth_network_tokenization(xml, payment_method, options)
-        return unless network_tokenization?(payment_method)
+        brand = card_brand(payment_method).to_sym
+        case brand
+        when :visa, :master, :american_express
```

Comment:
The documentation calls out additional supported brand types [here](https://developer.cybersource.com/docs/cybs/en-us/google-pay/developer/fdiglobal/rest/googlepay/googpay-intro/googpay-processors.html). Is there a reason we're not also adding support for Diners Club, JCB, or Discover here?